docs/modules/ROOT/pages/crosschain.adoc

```diff
@@ -0,0 +1,175 @@
+
+For example, we could have our token on xDai, with our governor on mainnet, or we could have our token on mainnet, with our governor on optimism
+
+In order to do that, we will start by adding xref:api:crosschain.adoc#CrossChainEnabled[`CrossChainEnabled`] to our contract.
+
+```diff
+import "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
++import "@openzeppelin/contracts-upgradeable/crosschain/CrossChainEnabled.sol";
+
+-contract MyToken is Initializable, ERC20Upgradeable, UUPSUpgradeable {
++abstract contract MyTokenCrossChain is Initializable, ERC20Upgradeable, UUPSUpgradeable, CrossChainEnabled {
+```
+
+Once that is done, we can use the `onlyCrossChainSender` modifier, provided by `CrossChainEnabled` in order to protect the minting and upgrading operations.
+
+```diff
+-    modifier onlyOwner() {
+-        require(owner == _msgSender(), "Not authorized");
+-        _;
+-    }
+
+-    function mint(address to, uint256 amount) public onlyOwner {
++    function mint(address to, uint256 amount) public onlyCrossChainSender(owner) {
+
+-    function _authorizeUpgrade(address newImplementation) internal override onlyOwner {
++    function _authorizeUpgrade(address newImplementation) internal override onlyCrossChainSender(owner) {
+```
+
+This change will effectively restrict the mint and upgrade operations to the `owner` on the remote chain.
+
+== Specializing for a specific chain
+
+Once the abstract cross-chain version of our token is ready we can easily specialize it for the chain we want, or more precisely for the bridge system that we want to rely on.
+
+This is done using one of the many `CrossChainEnabled` implementations.
+
+For example, if our token on xDai, and our governor on mainnet, we can use the https://docs.tokenbridge.net/amb-bridge/about-amb-bridge[AMB] bridge available on xDai at https://blockscout.com/xdai/mainnet/address/0x75Df5AF045d91108662D8080fD1FEFAd6aA0bb59[0x75Df5AF045d91108662D8080fD1FEFAd6aA0bb59]
+
+[source,solidity]
+----
+[...]
+
+import "@openzeppelin/contracts-upgradeable/crosschain/amb/CrossChainEnabledAMB.sol";
+
+contract MyTokenXDAI is
+    MyTokenCrossChain,
+    CrossChainEnabledAMB(0x75Df5AF045d91108662D8080fD1FEFAd6aA0bb59)
+{}
+----
+
+If token on mainnet, and our governor on optimism, we use the optimism https://community.optimism.io/docs/protocol/protocol-2.0/#l1crossdomainmessenger[CrossDomainMessenger] available on mainnet at https://etherscan.io/address/0x25ace71c97B33Cc4729CF772ae268934F7ab5fA1[0x25ace71c97B33Cc4729CF772ae268934F7ab5fA1]
+
+[source,solidity]
+----
+[...]
+
+import "@openzeppelin/contracts-upgradeable/crosschain/optimismCrossChainEnabledOptimism.sol";
+
+contract MyTokenOptimism is
+    MyTokenCrossChain,
+    CrossChainEnabledOptimism(0x25ace71c97B33Cc4729CF772ae268934F7ab5fA1)
+{}
+----
+
+== Going further with access control
```

Comment:
I think we should also explain how it can be dangerous to mix addresses from different chains. The code above with an owner is not a good example in that regard, so we should take some time to mention it and say how it could be improved. For example, it could be made explicit by changing the variable name to `l1Owner` or `crossChainOwner`, and we can use it as an opportunity to introduce `AccessControlCrossChain` as a way to avoid the problem also.